File: datasets/C++/Gemini/Algorithm/medium5.cpp
#include <iostream>

void solve() {
    int n;
    std::cin >> n;
    int xor_sum = 0;
    for (int i = 0; i < n; ++i) {
        long long m;
        int k;
        std::cin >> m >> k;
        
        int grundy_num = 0;
        if (m % 2 != 0) { // m is odd
            grundy_num = 1;
        } else { // m is even
            grundy_num = 2;
        }
        
        xor_sum ^= grundy_num;
    }
    
    if (xor_sum != 0) {
        std::cout << "BOB" << std::endl;
    } else {
        std::cout << "BEN" << std::endl;
    }
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    int g;
    std::cin >> g;
    while (g--) {
        solve();
    }
    return 0;
}

File: datasets/Python/Gemini/Algorithm/medium5.py
def solve():
    n = int(input())
    xor_sum = 0
    for _ in range(n):
        m, k = map(int, input().split())
        
        # Calculate Grundy number for the tree
        grundy_num = 0
        if m % 2 == 1: # m is odd
            grundy_num = 1
        else: # m is even
            grundy_num = 2
            
        xor_sum ^= grundy_num
        
    if xor_sum != 0:
        print("BOB")
    else:
        print("BEN")

g = int(input())
for _ in range(g):
    solve()